scripts/kfctl.sh

```diff
@@ -26,80 +26,103 @@ DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
 source "${DIR}/util.sh"
 source "${DIR}/gke/util.sh"
 source "${DIR}/util-minikube.sh"
+INPUT=''
+FORMAT=''
+export KUBEFLOW_COMPONENTS=${DEFAULT_KUBEFLOW_COMPONENTS:-'"ambassador","jupyter","centraldashboard","tf-job-operator","pytorch-operator","spartakus","argo","pipeline"'}
+export KUBEFLOW_EXTENDEDINFO=${KUBEFLOW_EXTENDEDINFO:-false}
+
+writeEnv() {
+  echo -e $INPUT | envsubst $FORMAT > ${ENV_FILE}
+}
 
 createEnv() {
   # Check if there is a file env.sh
   # If there is source it otherwise create it.
   # this ensures all relevant environment variables are persisted in
   # a file for consistency across runs.
-  echo PLATFORM=${PLATFORM} >> ${ENV_FILE}
+
   DEFAULT_KUBEFLOW_REPO="$(cd "${DIR}/.." > /dev/null && pwd)"
   # Remove trailing slash from the repo.
   KUBEFLOW_REPO=${KUBEFLOW_REPO%/}
-  echo KUBEFLOW_REPO=${KUBEFLOW_REPO:-"${DEFAULT_KUBEFLOW_REPO}"} >> ${ENV_FILE}
-  echo KUBEFLOW_VERSION=${KUBEFLOW_VERSION:-"master"} >> ${ENV_FILE}
-  echo KUBEFLOW_KS_DIR=${KUBEFLOW_KS_DIR:-"$(pwd)/ks_app"} >> ${ENV_FILE}
-  echo KUBEFLOW_DOCKER_REGISTRY=${KUBEFLOW_DOCKER_REGISTRY:-""} >> ${ENV_FILE}
-  echo DOCKER_REGISTRY_KATIB_NAMESPACE=${DOCKER_REGISTRY_KATIB_NAMESPACE:-""} >> ${ENV_FILE}
 
+  INPUT+='PLATFORM=$PLATFORM\nKUBEFLOW_REPO=$KUBEFLOW_REPO\nKUBEFLOW_VERSION=$KUBEFLOW_VERSION\n'
+  INPUT+="KUBEFLOW_COMPONENTS='$KUBEFLOW_COMPONENTS'\nKUBEFLOW_EXTENDEDINFO=$KUBEFLOW_EXTENDEDINFO\n"
+  INPUT+='KUBEFLOW_KS_DIR=$KUBEFLOW_KS_DIR\nKUBEFLOW_DOCKER_REGISTRY=$KUBEFLOW_DOCKER_REGISTRY\n'
+  INPUT+='DOCKER_REGISTRY_KATIB_NAMESPACE=$DOCKER_REGISTRY_KATIB_NAMESPACE\nK8S_NAMESPACE=$K8S_NAMESPACE\n'
+  INPUT+='KUBEFLOW_PLATFORM=$KUBEFLOW_PLATFORM\nMOUNT_LOCAL=$MOUNT_LOCAL\nDEPLOYMENT_NAME=$DEPLOYMENT_NAME\n'
+  FORMAT+='$PLATFORM$KUBEFLOW_REPO$KUBEFLOW_VERSION$KUBEFLOW_KS_DIR$KUBEFLOW_DOCKER_REGISTRY'
+  FORMAT+='$KUBEFLOW_COMPONENTS$DOCKER_REGISTRY_KATIB_NAMESPACE$K8S_NAMESPACE$KUBEFLOW_PLATFORM$MOUNT_LOCAL'
+  FORMAT+='$DEPLOYMENT_NAME$KUBEFLOW_EXTENDEDINFO'
+
+  export KUBEFLOW_REPO=${KUBEFLOW_REPO:-"${DEFAULT_KUBEFLOW_REPO}"}
+  export KUBEFLOW_VERSION=${KUBEFLOW_VERSION:-"master"}
+  export KUBEFLOW_KS_DIR=${KUBEFLOW_KS_DIR:-"$(pwd)/ks_app"}
+  export KUBEFLOW_DOCKER_REGISTRY=${KUBEFLOW_DOCKER_REGISTRY:-""}
+  export DOCKER_REGISTRY_KATIB_NAMESPACE=${DOCKER_REGISTRY_KATIB_NAMESPACE:-""}
   # Namespace where kubeflow is deployed
-  echo K8S_NAMESPACE=${K8S_NAMESPACE:-"kubeflow"} >> ${ENV_FILE}
+  export K8S_NAMESPACE=${K8S_NAMESPACE:-"kubeflow"}
 
   case "$PLATFORM" in
     minikube)
-      echo KUBEFLOW_PLATFORM=minikube >> ${ENV_FILE}
-      echo MOUNT_LOCAL=${MOUNT_LOCAL} >> ${ENV_FILE}
+      export KUBEFLOW_PLATFORM=minikube 
```

Comment:
Why did you change this to an export statement rather than writing to a file?

The reason to write to a file was so that we can reuse the same values on subsequent invocations.

Reply:
See above comment about envsubstr and env source files

Reply:
Much nicer thanks